logstash-core/src/test/java/org/logstash/FileLockFactoryMain.java

```diff
@@ -0,0 +1,23 @@
+package org.logstash;
+
+import java.io.IOException;
+
+/*
+ * This program is used to test the FileLockFactory in cross-process/JVM.
+ */
+public class FileLockFactoryMain {
+
+    public static void main(String[] args) {
+        // Timeout to ensure the process finishes.
+        long timeout = 20000;
+        try {
+            FileLockFactory.getDefault().obtainLock(args[0], args[1]);
+            System.out.println("File locked");
+            Thread.sleep(timeout);
```

Comment:
@wainersm added a suggestion about safety below, I agree though Java sub-processes are never optimal :( But I think the JVM just dying without the finally block being hit is unlikely enough to not care about here :)